tsconfig.json

```diff
@@ -0,0 +1,14 @@
+{
```

Comment:
Do we need to include source maps to get accurate code coverage on the typescript source from nyc?

Reply:
No idea! I haven’t yet the pleasure of running coverage with TS.

I can say that in my experience coverage is relatively less important, especially compared to the “does this code run or not” stage we’ve currently reached in the frontend.

Reply:
Not that I'm necessarily arguing _against_ testing, just that I think the wins from typing are big, and while I think most behaviors in the frontend will still need to be tested (and manually), TS React codebases are much less at risk of breaking during refactors or unrelated changes. I've never had a React app with a stable enough structure that the test-writing paid off, especially compared to the benefit from compile-time type checking.

Reply:
Understood. I only mentioned it since we're currently capturing and publishing code coverage, including the frontend, to coveralls. 

We don't need to block, but I would like to see that we continue to produce coverage for the frontend (and can hopefully avoid our project-wide coverage of 93% dropping 😄)

Reply:
I think what's happening right now is that no coverage is reported for the TS files. I just tried for a few minutes but didn't manage to get it working. 

Could be worth opening an issue with https://github.com/piotrwitek/ts-mocha which is the way we're running tests (the easiest way I've found) but doesn't have any mention of using it with istanbul/nyc.

Alternatively I found these resources, not using `ts-mocha`. Probably the place to start on those would be to remove `ts-mocha` and try to get the tests running again.

- https://azimi.me/2016/09/30/nyc-mocha-typescript.1.html
- https://istanbul.js.org/docs/tutorials/typescript/
- https://github.com/john-cheesman/ts-mocha-chai-nyc

You're saying we don't need to block… should we open a new issue?

Reply:
> You're saying we don't need to block… should we open a new issue?

Will defer to you. If there's a big value add in getting TS for the frontend sooner rather than later, then I don't want to hold that up while we try to figure out the coverage story. 

I've always used istanbul/nyc with mocha, but also transpile beforehand via tsc. I'll check out this branch locally and poke around the code coverage/tests as well.

Reply:
Yea, I'm itching to use this 😁

My inclination would be to drop frontend coverage until it feels pressing, in part because I continue to be surprised at how difficult it is to get babel/ts/mocha/istanbul happy with each other. I spent a surprising amount of time getting it working in the first place… it's frustrating. Coverage should be easier to accomplish than it is.

I'm also happy to leave it in place for the part that's still in .js and hopefully get around to fixing it sooner rather than later. Or to wait a little bit if you want to poke around!

Reply:
>My inclination would be to drop frontend coverage until it feels pressing

> I'm also happy to leave it in place for the part that's still in .js and hopefully get around to fixing it sooner rather than later. Or to wait a little bit if you want to poke around!

I'm sold! Let's deal with coverage in a follow on. I'll open a new issue after this gets merged